pkg/kubelet/kubelet.go

```diff
@@ -2117,66 +2117,71 @@ func (kl *Kubelet) syncNetworkStatus() {
+	if err != nil {
+		glog.Errorf("Failed to get addresses from cloud provider, so node addresses will be stale: %v", err)
+		return
+	}
+	node.Status.Addresses = nodeAddresses
+}
+
+func (kl *Kubelet) setUpdatedAddressesFromHostname(node *api.Node) {
+	addr := net.ParseIP(kl.hostname)
+	if addr == nil {
+		addrs, err := net.LookupIP(node.Name)
+
 		if err != nil {
-			return fmt.Errorf("failed to get node address from cloud provider: %v", err)
+			glog.Errorf("can't get ip address of node %s, so node addresses will be stale: %v", node.Name, err)
+			return
 		}
-		node.Status.Addresses = nodeAddresses
-	} else {
-		addr := net.ParseIP(kl.hostname)
-		if addr != nil {
-			node.Status.Addresses = []api.NodeAddress{
-				{Type: api.NodeLegacyHostIP, Address: addr.String()},
-				{Type: api.NodeInternalIP, Address: addr.String()},
-			}
-		} else {
-			addrs, err := net.LookupIP(node.Name)
-			if err != nil {
-				return fmt.Errorf("can't get ip address of node %s: %v", node.Name, err)
-			} else if len(addrs) == 0 {
-				return fmt.Errorf("no ip address for node %v", node.Name)
-			} else {
-				// check all ip addresses for this node.Name and try to find the first non-loopback IPv4 address.
-				// If no match is found, it uses the IP of the interface with gateway on it.
-				for _, ip := range addrs {
-					if ip.IsLoopback() {
-						continue
-					}
-
-					if ip.To4() != nil {
-						node.Status.Addresses = []api.NodeAddress{
-							{Type: api.NodeLegacyHostIP, Address: ip.String()},
-							{Type: api.NodeInternalIP, Address: ip.String()},
-						}
-						break
-					}
-				}
 
-				if len(node.Status.Addresses) == 0 {
-					ip, err := util.ChooseHostInterface()
-					if err != nil {
-						return err
-					}
+		if len(addrs) == 0 {
+			glog.Errorf("no ip address for node %v, so node addresses will be stale", node.Name)
+			return
+		}
 
-					node.Status.Addresses = []api.NodeAddress{
-						{Type: api.NodeLegacyHostIP, Address: ip.String()},
-						{Type: api.NodeInternalIP, Address: ip.String()},
-					}
-				}
+		// check all ip addresses for this node.Name and try to find the first non-loopback IPv4 address.
+		// If no match is found, it uses the IP of the interface with gateway on it.
+		for _, ip := range addrs {
+			if !ip.IsLoopback() && ip.To4() != nil {
+				addr = ip
+				break
 			}
 		}
+
+		if addr == nil {
+			ip, err := util.ChooseHostInterface()
+			if err != nil {
+				glog.Errorf("Failed choosing host interface, so node addresses will be stale: %v", err)
+				return
+			}
+			addr = ip
+		}
+	}
+
+	node.Status.Addresses = []api.NodeAddress{
+		{Type: api.NodeLegacyHostIP, Address: addr.String()},
+		{Type: api.NodeInternalIP, Address: addr.String()},
+	}
+}
+
+// setNodeStatus fills in the Status fields of the given Node, overwriting
+// any fields that are currently set.
+func (kl *Kubelet) setNodeStatus(node *api.Node) error {
+
+	// Set addresses for the node.
```

Comment:
Please update this comment to explain that the address may be stale if there is an issue getting a new value (since no errors from the following two functions are propagated back up).